pkg/sql/delegate/show_regions.go

```diff
@@ -28,31 +27,42 @@ func (d *delegator) delegateShowRegions(n *tree.ShowRegions) (tree.Statement, er
 	}
 	sqltelemetry.IncrementShowCounter(sqltelemetry.RegionsFromCluster)
 
-	// TODO (storm): Change this so that it doesn't use hard-coded strings and is
-	// more flexible for custom named sub-regions.
+	regionKey := "region"
+	zoneKeys := []string {"zone", "az", "availability_zone"}
+	zoneKeysSubClause := ""
+
+	// Build the zone sub-clause based on the zone keys that we're accepting above.
+	for i, zk := range zoneKeys {
+		zoneKeysSubClause += `
```

Comment:
i concur - it's harder to read (and slightly more inefficient)
i kind of think this belongs in a generation script rather than hardcoded in this way.
i'm not opposed to it, but if you asked for my preferences it would be to keep it as is.